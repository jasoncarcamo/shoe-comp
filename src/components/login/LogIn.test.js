import React from "react";
import ReactDOM from "react-dom";
import {BrowserRouter} from "react-router-dom";
import Login from "./LogIn";
import renderer from "react-test-renderer";
import Enzyme, {shallow} from "enzyme";
import Adaptor from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adaptor()});


describe("<Login/>", ()=>{
    it("<Renders without crashing", ()=>{
        const div = document.createElement("div");

        ReactDOM.render(<BrowserRouter><Login/></BrowserRouter>, div);

        ReactDOM.unmountComponentAtNode(div);
    });

    it("Creates a snapshot", ()=>{
        const tree = renderer.create(<BrowserRouter><Login/></BrowserRouter>).toJSON();

        expect(tree).toMatchSnapshot();
    });
});