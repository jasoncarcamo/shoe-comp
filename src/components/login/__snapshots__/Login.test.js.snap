// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login/> Creates a snapshot 1`] = `
<section
  id="login-section"
>
  <form
    id="login-form"
    onSubmit={[Function]}
  >
    <fieldset>
      <label
        htmlFor="login-email"
      >
        Email: 
      </label>
      <input
        id="login-email"
        onChange={[Function]}
        type="email"
      />
      <label
        htmlFor="login-password"
      >
        Password:
      </label>
      <input
        id="login-password"
        onChange={[Function]}
        type="password"
      />
      <span
        id="login-error"
      >
        Not registered? Sign up 
        <a
          href="/register"
          onClick={[Function]}
        >
          here
        </a>
      </span>
      <button
        id="login-submit"
        type="submit"
      >
        Log In
      </button>
    </fieldset>
  </form>
</section>
`;
